javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll

```diff
@@ -513,13 +513,24 @@ module TaintedPath {
 
     override predicate blocks(boolean outcome, Expr e) {
       member = "relative" and
-      e = pathCall.getArgument(1).asExpr() and
+      e = maybeGetJoinArg(pathCall.getArgument(1)).asExpr() and
       outcome = startsWith.getPolarity().booleanNot()
       or
       not member = "relative" and
-      e = pathCall.getArgument(0).asExpr() and
+      e = maybeGetJoinArg(pathCall.getArgument(0)).asExpr() and
       outcome = startsWith.getPolarity()
     }
+
+    bindingset[e]
+    private DataFlow::Node maybeGetJoinArg(DataFlow::Node e) {
```

Comment:
I'm having trouble relating the name and the semantics of this predicate. Could you add a bit of qldoc? Alternative names, based on semantics: `maybeGetLastJoinArg`, `getPathSuffix`?

Reply:
I ended up with `maybeGetPathSuffix`.  
I want the `maybe` in there to capture that the predicate also acts as the identity function.